### ISSUE TITLE
iOS not supported 

### ISSUE THREAD

**Issue by dataenthusiast-io** [B4PT0R/streamlit-mic-recorder#17]
I have embedded the recorder as per documentation into my streamlit application:

```
import streamlit as st
from streamlit_mic_recorder import mic_recorder
audio = mic_recorder(start_prompt='Start Recording', stop_prompt='Stop Recording', key='recorder')
st.audio(audio['bytes'])
```

The recording works fine across all major browser incl. Chrome, Firefox it works perfectly fine. Even for Android mobile devices I can properly use the recording feature. Nevertheless, in Safari and iOS in particular, it seems that now audio is recorded at all. It asks for permissions which can be granted but there does not seem to come any signal through.

As this is such a great addition to any kind of future AI / LLM app, it would be awesome to further investigate and fix this.

**Comment by sairaghavendravenkat** [comment]
Yes i also have this issue from last 6months. Please provide help @B4PT0R

**Comment by kodjac** [comment]
For whatever reason, it works for me on iOS with Safari using WAV format instead of the default WEBM format:

 `mic_recorder(..., format="wav")`

**Comment by manuelrech** [comment]
It does not work for me either. Any solution on this?

**Comment by manuelrech** [comment]
I understood the issue. When you run a streamlit app it runs on http. iOS gives mic permissions only to https sites.

You can switch to https using ngrok
